### ISSUE TITLE
opencv solver for 5-shot evaluation

### ISSUE THREAD

**Issue by Tsingularity** [icoz69/DeepEMD#32]
Hi,

I saw the default setting (both the eval.py and the evaluation command-line script in the ReadMe file) is using opencv solver. However, for 5-shot evaluation, the structured FC weights need gradients to update. But looks like the opencv solver layer is not differentiable? So how could the SFC weights be updated given this situation?

Thanks!

**Comment by karayzhou** [comment]
I'm also confused about the opencv solver layer, is this layer differentiable.

**Comment by icoz69** [comment]
hi 
in the 5-shot case, the gradient can flow back to the prototypes through the cost matrix.

**Comment by karayzhou** [comment]
thanks for your reply, I'm very interested in your execllent work. could provide more training details about 5-shot case, such as lr, max_epoch,val_episode,sfc_lr,and so on.
in addition, in 5-shot case ,is it necessary to pretrain the model in 5-case scene, that is in validation phase, select the best model in 5-shot Deepemd setting.

**Comment by icoz69** [comment]
hi, 5-shot testing does not need training. you can simply use the trained one-shot model to undertake 5-shot testing and use the default parameters.  carefully tuning the sfc_lr can obtain better results.
